main.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/caarlos0/env"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+)
+
+type config struct {
+	GithubToken string `env:"GHTOKEN"`
+	Authors     string `env:"AUTHORS"`
+}
+
+type UserData struct {
+	PrCount   int
+	AvatarURL string
+}
+
+func leaderboard(writer http.ResponseWriter, request *http.Request) {
+	userData := make(map[string]UserData)
+	authors := strings.Split(cfg.Authors, ":")
+	fmt.Printf("Authors: %v\n", authors)
+	for _, author := range authors {
+		userData[author] = UserData{PrCount: getPrCount(author), AvatarURL: getAvatar(author)}
+		fmt.Printf("Author: %s, PR count: %d\n", author, userData[author].PrCount)
+	}
+	jsonString, _ := json.Marshal(userData)
+	fmt.Fprintf(writer, "%s", jsonString)
+}
+
+func getAvatar(author string) string {
+	url := fmt.Sprintf("https://api.github.com/users/%s", author)
+	response, err := http.Get(url)
+	if err != nil {
+		fmt.Println("Failed to fetch avatar. %s\n", err)
+		return ""
+	} else {
+		ghData, _ := ioutil.ReadAll(response.Body)
+		var result map[string]interface{}
+		json.Unmarshal([]byte(ghData), &result)
+		return result["avatar_url"].(string)
+	}
+}
+
+func getPrCount(author string) (prCount int) {
+	defer func() {
+		if r := recover(); r != nil {
+			fmt.Println("recovering from panic. %v", r)
+			prCount = -1
+		}
+	}()
+
+	year := calcYear()
+	url := fmt.Sprintf("https://api.github.com/search/issues?q=created:%d-09-30T00:00:00-12:00..%d-10-31T23:59:59-12:00+type:pr+is:public+author:%s", year, year, author)
+	client := &http.Client{}
+	request, _ := http.NewRequest("GET", url, nil)
+	if cfg.GithubToken != "" {
+		request.Header.Set("Authorization", "token "+cfg.GithubToken)
+	}
+	response, err := client.Do(request)
+	if err != nil {
+		fmt.Println("Failed to fetch PR count. %s\n", err)
+		return -1
+	} else {
+		ghData, _ := ioutil.ReadAll(response.Body)
+		var result map[string]interface{}
```

Comment:
why don't you define the struct expected with the field for total count as type int?

Reply:
Did the same thing here.